test/verify/check-connection

```diff
@@ -966,6 +966,51 @@ until pgrep -f '^/usr/bin/cockpit-bridge.*--privileged'; do sleep 1; done
         m.write("/etc/cockpit/ws-certs.d/0-self-signed-ca.pem", "FAKE CERT FOR TESTING\n")
         self.assertEqual(m.execute("curl -sfS http://localhost:9090/ca.cer"), "FAKE CERT FOR TESTING\n")
 
+    def test_branding(self):
+        m = self.machine
+
+        # for all of our CI images, the part before the dash is the name of the
+        # subdirectory in /usr/share/cockpit/branding.
+        brand = m.image.split('-')[0]
+        branddir = f'/usr/share/cockpit/branding/{brand}'
+
+        # make sure that there are no broken links in "our" directory
+        self.assertFalse(m.execute(f'find {branddir} -xtype l'))
+
+        # branding.css undergoes variable substitution based on the content of
+        # /usr/lib/os-release.  Perform the substitution for ourselves for
+        # validation.  envsubst comes from gettext.
+        os_release_vars = m.execute("cat /usr/lib/os-release").replace('\n', ' ')
+        m.execute(f'{os_release_vars} envsubst < {branddir}/branding.css > /tmp/branding.ref')
+
+        # fetch some files and make sure they match what we expect
+        def curl_and_compare(name, content_type=None, reference=None):
+            url = f'http://localhost:9090/cockpit/static/{name}'
+            reference = reference or f'{branddir}/{name}'
+
+            # Check that the expected content type is served
+            if content_type is not None:
+                self.assertIn(f'Content-Type: {content_type}', m.execute(f'curl --head {url}'))
+
+            # Check that we can fetch the file
+            m.execute(f'curl --fail -o /tmp/{name} {url}')
+
+            # compare that it matches what we expected
+            m.execute(f'cmp /tmp/{name} {reference}')
+
+        m.start_cockpit()
+
+        # some brands miss the images, but the OSes we CI have them all
+        curl_and_compare('branding.css', 'text/css', reference='/tmp/branding.ref')
```

Comment:
I'm honestly not a fan about this. It's prodding in internal implementation details, and integration tests ought to test things that are visible/relevant to the user. Like the pixel test, which already makes sure that the logo works.

No veto, but unnecessary complexity IMHO.

Reply:
The main reason I like these "other" tests is that they run on other OSes, whereas pixel testing only makes sure we didn't break Fedora.

For something that's very specifically designed to be different on different OSes, I think this is worth testing.  We might find out on image udpate that some png that we were using for the logo has disappeared from the OS, for example, which is useful information.

Reply:
Right, fair enough.